Gems/Atom/Bootstrap/Code/Source/BootstrapSystemComponent.cpp

```diff
@@ -391,6 +393,83 @@ namespace AZ
 
                 AZ::RHI::MultisampleState multisampleState;
 
+                if (xrSystem)
+                {
+                    RHI::Device* device = RHI::RHISystemInterface::Get()->GetDevice();
+                    if (RHI::CheckBitsAll(device->GetFeatures().m_shadingRateTypeMask, RHI::ShadingRateTypeFlags::PerRegion) &&
+                        !m_vrsTexture)
+                    {
+                        // Need to fill the contents of the Variable shade rating image.
+                        const AZStd::shared_ptr<const RPI::PassTemplate> forwardTemplate =
+                            RPI::PassSystemInterface::Get()->GetPassTemplate(Name("MultiViewForwardPassTemplate"));
```

Comment:
This limits the vrs system to only the setup that was selected when the application is launched. Is that planned to be addressed later on?

I'd also question if the bootstrap system component is the best place for this to reside, seems like it could be part of the RPI?

Reply:
It would be nice to do this work within RPI instead of here and ASV as well. Thoughts on adding this code to RasterPass? Maybe you can override Pass::BuildInternal for Rasterpass and add this code in there. Tagging @VickyAtAZ  in case she has a better suggestion.

Reply:
I dont have a better place to move it to within RPI (unless someone else has a better solution) and I do not want to block this work for this reason. We can add a todo comment for now.